### PR TITLE
Fix locking

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

File: caps/repo.go [hunk 1]
```diff
@@ -81,6 +81,14 @@ func (r *Repository) HasType(t *Type) bool {
 	return false
 }
 
+// HasType checks if the repository contains the given type.
```
Comment (chipaca) [inline caps/repo.go]: checks *whether*, not if.

**Comment by chipaca** [comment]
:+1:

File: caps/repo.go (hunk 1 above)
Comment (zyga) [inline caps/repo.go]: Noted, I'll fix this in the next round. I have a few more Foo() == foo() + locking methods in the pipe

File: caps/repo.go [hunk 2]
```diff
@@ -71,8 +71,8 @@ func (r *Repository) Add(cap *Capability) error {
 	return nil
 }
 
-// HasType checks if the repository contains the given type.
-func (r *Repository) HasType(t *Type) bool {
+// hasType is identical to HasType but doesn't hold the repository lock
```
Comment (niemeyer) [inline caps/repo.go]: And you expect we, wannabe robots, to never mistype h by H or the opposite on every context? :-)

This should be named unlockedHasType or something. Or, alternatively, can we get rid of the public version altogether? I can't see many practical public use cases for a full-type comparison like this.

**Comment by niemeyer** [comment]
Still pending an answer to whether we can drop the public version, or whether we should rename it.

File: caps/repo.go (hunk 2 above)
Comment (zyga) [inline caps/repo.go]: +1 on unlockedFoo when there's a need for having both. I'll kill the public version in this case as you are right it is not really needed.

File: caps/repo.go (hunk 1 above)
Comment (zyga) [inline caps/repo.go]: Fixed

**Comment by mvo5** [comment]
Sorry, I'm late to the party and now I'm confused. The branch says "adds locking" but I do not see locking anymore in the diff? Is github lying to me? Or did something else change here?

**Comment by zyga** [comment]
The branch evolved to the point where loccking is gone as the only remaining method is private and you are expected to call it from public methods that already take the lock

**Comment by mvo5** [comment]
Thanks, in this case +1 from me

**Pull request merged**